### DISCUSSION TITLE
Unable to use the value variable inside array

### DISCUSSION THREAD

**Discussion by videni** [statamic/cms#8225] [Troubleshooting] [answered]
Statamic version: 4.1.1

```
interiors:
  - colorker-verona-6.webp
  - shd_start-naxos-12.webp
  - shd_start-naxos-13.webp
  - shd_start-naxos-3.webp
favorites: 1234567890
```

```
{{ favorites}}
{{ interiors }}
        Value: {{ value }}
        Name:  {{ index }}
        Favorites: {{ favorites}}
{{/interiors}}
```

The interiors variable is an image assets which has favorites field.   

 I can get the index, but not value, the reason why I have to use the value, because the parent "favorites" variable  will override the nested one. the usage is  from offical doc, it  seems it is outdated.    any one know a workaround please?

**Comment by jasonvarga** [comment, marked as answer]
If the asset has its own `favourites` variable, it should override the outer one.

I'm assuming where you are seeing the outer one, it's because that particular asset in the loop doesn't have its own `favourites`.

You can get around this by using a scope.

```antlers
{{ favourites }}

{{ interiors scope="interior" }}
  {{ interior:favourites }}
{{ /interiors }}
```

By using the `scope`, each iteration will be inside the given variable (`interior`). So when you do `{{ interior:favourites }}`, it will _only_ get it from inside `interior` and won't fall back to the parent.

**Reply by videni** [reply]
thanks ,it works like a charm.   I didn't find the doc about the `scope` in the offical site,   can you also show me the link please?
